src/os/explorer/components/Topbar.tsx

```diff
@@ -279,14 +256,31 @@ export const Topbar: React.FC<TopbarProps> = ({
               Make a copy
             </DropdownMenuItem>
 
-            <DropdownMenuItem onClick={() => exportAsMarkdown()}>
-              <Download size={14} className="inline-block text-gray-500 mr-2" />{" "}
-              Export as Markdown
-            </DropdownMenuItem>
-            <DropdownMenuItem onClick={() => downloadAsAutomerge()}>
-              <SaveIcon size={14} className="inline-block text-gray-500 mr-2" />{" "}
-              Download Automerge file
-            </DropdownMenuItem>
+            {(selectedDatatypeMetadata?.fileExportMethods ?? [])
+              .concat(genericExportMethods)
```

Comment:
I think we should probably make datatypes into classes which inherit from a base class. That might help with things like this since the base class could add generic export methods to datatype-specific ones. The base class could also have default behaviors on it—currently we have a bunch of optional fields on the `Datatype` POJO and then we fill in defaults throughout the app which seems bad.